pkg/gardenlet/controller/factory.go

```diff
@@ -135,9 +135,9 @@ func (f *GardenletControllerFactory) Run(ctx context.Context) error {
 		return fmt.Errorf("timed out waiting for Kube caches to sync")
 	}
 
-	seeds := seedNames(f.cfg.SeedConfig, f.k8sGardenCoreInformers.Core().V1beta1().Seeds().Lister(), f.cfg.SeedSelector)
-	if len(seeds) < 1 {
-		return fmt.Errorf("no seed selected by Gardenlet")
+	gardenSecretsNamespace := v1beta1constants.GardenNamespace
```

Comment:
I'm afraid that this won't work in the future as Gardenlet will not be permitted to read secrets from the `garden` namespace.

Reply:
I also have some doubts about this, but if gardenlet is not allowed to read secrets from `garden` namespace then it will simply fail with authorization error. I think this should be fine, wdyt?

Reply:
Hm, I would rather favor skipping the check. Reading `Secrets` from the `garden` namespace is something we don't expect to work, so we probably shouldn't try to do it in the first place (also for sake of code maintenance).

Reply:
Thanks for the feedback! I have removed this secret sync now.

Reply:
Thanks for changing it.